src/revlanguage/monitors/Mntr_StochasticBranchRate.cpp

```diff
@@ -69,6 +70,19 @@ void Mntr_StochasticBranchRate::constructInternalObject( void )
         m->setAppend( app );
         m->setPrintVersion( wv );
     }
+    if ( static_cast<const RevLanguage::Tree&>( bds->getRevObject() ).isModelObject() )
+    {
+        RevBayesCore::TypedDagNode<RevBayesCore::Tree>* bds_tdn = static_cast<const RevLanguage::Tree&>( bds->getRevObject() ).getDagNode();
+        RevBayesCore::StochasticNode<RevBayesCore::Tree>* bds_sn  = static_cast<RevBayesCore::StochasticNode<RevBayesCore::Tree>* >( bds_tdn );
+
+        RevBayesCore::FastBirthDeathShiftProcess *bds_process = NULL;
+        bds_process = dynamic_cast<RevBayesCore::FastBirthDeathShiftProcess*>( &bds_sn->getDistribution() );
+        bds_process->setSampleCharacterHistory( true );
+
+        m = new RevBayesCore::StochasticBranchRateMonitor( bds_sn, (std::uint64_t)print_gen, file_name, sep );
+        m->setAppend( app );
+        m->setPrintVersion( wv );
+    }
```

Comment:
Since we are adding a third valid option in addition to CDBDP and GLHBDSP, the exception thrown in the `else` branch should be updated accordingly.